### PR TITLE
Modified picasso tests to work with CI

### PR DESCRIPTION
**Pull request by chemistry-sourabh**
Modified Picasso Tests to run with CI #133

### CODE REVIEW

**Comment by chemistry-sourabh** [comment]
Unfortunately the Coverage is not generated for Picasso and Einstein since they are created as different processes. Will try looking into this.

File: tests/unit/picasso/test_rest.py [hunk 1]
```diff
@@ -35,6 +39,25 @@
...
+
+def setUpModule():
+    global processes
+    p1 = Process(target=start_name_server)
+    p2 = Process(target=start_rpc_server)
+    p3 = Process(target=rest.start)
+    p1.start()
+    p2.start()
+    p3.start()
+    time.sleep(5)
```
Comment (naved001) [inline tests/unit/picasso/test_rest.py]: just a minor nit: you could start the processes the same way you are terminating itthem: `[p.start() for p in processes]`
You could leave it as it is if there are some other reasons.

**Comment by naved001** [comment]
@CCI-MOC/bmi-breakers we need one more reviewer.

**Comment by ravisantoshgudimetla** [comment]
I am sorry, I didn't understand the issue. I need some more background on this before reviewing. Feel free to choose someone else as reviewer.

**Comment by chemistry-sourabh** [comment]
@ravisantoshgudimetla nothing much. Picasso Tests were not enabled in CI as they need running of Picasso and Einstein processes. So added starting and stopping of Einstein and Picasso processes in the module so that it will work in CI and also make it easier for programmers to run these tests as they will not need to run Picasso and Einstein before hand.

**Comment by chemistry-sourabh** [comment]
@naved001 done!

**Comment by naved001** [comment]
@chemistry-sourabh Can you explain what was the coverage issue and how you fixed it? In fact, it would be better if you put that in the commit message. Thanks!

**Comment by chemistry-sourabh** [comment]
@naved001 sadly I didn't fix it yet. The issue is that the coverage is not collected for the rpc_server, name_server and Picasso_server as they are started as different processes. Trying to get tests to dump coverage for them so that we will have accurate measurement.

**Comment by chemistry-sourabh** [comment]
@naved001 fixed the coverage issue. Ready for review

**Comment by naved001** [comment]
@chemistry-sourabh you da best. I'll ask questions tomorrow.

**Comment by chemistry-sourabh** [comment]
@naved001 added the long explanation.

**Comment by naved001** [comment]
@djfinn14 @ravisantoshgudimetla @jeremyfreudberg either of you want to review this?

File: tests/unit/picasso/test_rest.py [hunk 2]
```diff
@@ -35,6 +39,47 @@
...
+    stop_services = False
+    threads = [Thread(target=start_name_server),
+               Thread(target=start_rpc_server),
+               Thread(target=rest.start)]
+    for t in threads:
+        t.daemon = True
+        t.start()
+
+    while not stop_services:
+        time.sleep(0)
```
Comment (jeremyfreudberg) [inline tests/unit/picasso/test_rest.py]: curious about `time.sleep(0)` versus `pass`, or any other choice of no-op statement. not sure if it really matters...

Comment (chemistry-sourabh) [inline tests/unit/picasso/test_rest.py]: time.sleep(0) will result in context switch whereas pass may or may not do so.

Comment (jeremyfreudberg) [inline tests/unit/picasso/test_rest.py]: Cool! then i definitely approve!

**Comment by naved001** [comment]
2 signoffs, so merging

**Pull request merged**